### DISCUSSION TITLE
V2.10.0 - > NO SD CARD message with small SD Card (512Mb, 1Gb)

### DISCUSSION THREAD

**Discussion by OlivierBoubou** [EdgeTX/edgetx#5007] [Q&A]
After updating Edgetx to version 2.10.0, I have the message NO SD CARD with small capacity SD cards (512Mb, 1Gb)

**Comment by mhk1058** [comment]
Yes, I had that. It worked eventually but then intermittently. I didn't associate it with the card size. Eventually, trying to get it to connect, I got the message 'preparing storage' and my radio was cleared back to default. I managed to get it to connect and restored it however a few other issues, posted here earlier, resulted in me going back to 2.9.6.
i) Volume setting doesn't load ahead of welcome message, so first second or so is full volume regardless of actual setting
ii) Power light starts blue but turns green once radio is fully booted. Green was always just charging, blue powered on. Now I don't care what colour the power light is (although I do like blue) but previously green indicated a different state for the radio and until I know this is intentional I'm not risking it.
iii) SD card recognition issues as you mention

I know first reaction may well be, ah, you messed up the flash, however I have done this quite a few times now all the way back to Black Pearl with no issues. A few others on a TX16s Facebook group have reported the green power light too.
All OK now running 2.9

**Reply by Gaylian** [reply]
Same problem here (no SD card).
My real issue is why is the firmware for not supported Tx's (FrSky X9D / X9D+ / X9D+ SE (since v2.4)) being put in the V2.10.0 firmware files ?
If their not supported why have them there ? They don't work.
EdgeTX "Providence" v2.9.4 works OK

**Reply by pfeerick** [reply]
Please don't connect two unrelated issues. The Frsky family of radio is only community supported - there is no manufacturer support provided to the EdgeTX developers, and only some of us have the hardware. 

What you are referring to as a "small SD card" issue is most likely due to timing changes that occurred in the SD card drivers in 2.10, where it is actually a "slow SD card" issue whereby the SD card is slow/unresponsive to the initialisation request on powerup within the timeout period. We did see that earlier on in the development cycle for 2.10, and extended the timeout to resolve that for the cards we readily have access to. Perhaps these cards need even longer still. 

So #5015 will probably track this issue. 

> ii) Power light starts blue but turns green once radio is fully booted. Green was always just charging, blue powered on. Now I don't care what colour the power light is (although I do like blue) but previously green indicated a different state for the radio and until I know this is intentional I'm not risking it.

See https://github.com/EdgeTX/edgetx/issues/5010. It is only cosmetic, nothing to be concerned about.

**Reply by Gaylian** [reply]
Tried a newer SD Card and V2.10.0 works on a Frsky X9D. Should have done that sooner before I had a rant.

**Reply by pfeerick** [reply]
Thanks for the update, and it is perfectly understandable... it is frustrating when something that worked suddenly stops working on an update. Any idea of a brand or anything for the old SD card, or is it just a generic?

**Reply by Gaylian** [reply]
The card is the original one that came with the X9D. It is a Toshiba 2GB Micro SD made in Taiwan.
